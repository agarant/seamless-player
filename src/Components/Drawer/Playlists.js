import React from "react";
import List, { ListItem, ListItemIcon, ListItemText } from "material-ui/List";
import PlayListIcon from "material-ui-icons/PlaylistPlay";

export default ({ playlists, selectPlaylist }) => (
  <List>
    <div>
      {Object.keys(playlists).map(key => (
        <ListItem key={key} button onClick={() => selectPlaylist(key)}>
          <ListItemIcon>
            <PlayListIcon />
          </ListItemIcon>
          <ListItemText primary={playlists[key].name} />
        </ListItem>
      ))}
    </div>
  </List>
);
