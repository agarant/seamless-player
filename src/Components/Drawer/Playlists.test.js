import React from "react";
import renderer from "react-test-renderer";
import Playlists from "./Playlists";

test("<Playlists /> renders no list item when the playlists object is empyt", () => {
  const tree = renderer.create(<Playlists playlists={{}} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("<Playlists /> renders the playlist names", () => {
  const mockPlaylists = {
    "1": {
      name: "Blogged 50"
    },
    "2": {
      name: "Thumbs up"
    }
  };
  const tree = renderer
    .create(<Playlists playlists={mockPlaylists} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
