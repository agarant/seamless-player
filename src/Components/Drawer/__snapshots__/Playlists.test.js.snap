// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Playlists /> renders no list item when the playlists object is empyt 1`] = `
<ul
  className="MuiList-root-1 MuiList-padding-2"
>
  <div />
</ul>
`;

exports[`<Playlists /> renders the playlist names 1`] = `
<ul
  className="MuiList-root-1 MuiList-padding-2"
>
  <div>
    <li
      className="MuiButtonBase-root-15 MuiListItem-root-5 MuiListItem-gutters-12 MuiListItem-button-13 MuiListItem-default-8"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-18 MuiListItemIcon-root-17"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 9H2v2h17V9zm0-4H2v2h17V5zM2 15h13v-2H2v2zm15-2v6l5-3-5-3z"
        />
      </svg>
      <div
        className="MuiListItemText-root-19"
      >
        <h3
          className="MuiTypography-root-24 MuiTypography-subheading-31 MuiListItemText-text-22"
        >
          Blogged 50
        </h3>
      </div>
      <span
        className="MuiTouchRipple-root-48"
      />
    </li>
    <li
      className="MuiButtonBase-root-15 MuiListItem-root-5 MuiListItem-gutters-12 MuiListItem-button-13 MuiListItem-default-8"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-18 MuiListItemIcon-root-17"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 9H2v2h17V9zm0-4H2v2h17V5zM2 15h13v-2H2v2zm15-2v6l5-3-5-3z"
        />
      </svg>
      <div
        className="MuiListItemText-root-19"
      >
        <h3
          className="MuiTypography-root-24 MuiTypography-subheading-31 MuiListItemText-text-22"
        >
          Thumbs up
        </h3>
      </div>
      <span
        className="MuiTouchRipple-root-48"
      />
    </li>
  </div>
</ul>
`;
